File: iprep/strings/making_anagrams.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the makeAnagram function below.
def makeAnagram(a, b):
    dels = 0

    a_count = count_chars(a)
    b_count = count_chars(b)


    dels += count_dels(a_count, b_count)
    dels += count_dels(b_count, a_count)

    return dels

def count_chars(a):
    a_count = {}

    for c in a:
        if c in a_count: a_count[c]+=1
        else: a_count[c] = 1

    return a_count

def count_dels(a_count, b_count):
    dels = 0
    for c in a_count.keys():
        if c in b_count:
            if a_count[c]>b_count[c]:
                dels += a_count[c]-b_count[c]
                a_count[c] = b_count[c]
        else:
            dels += a_count[c]
            a_count[c] = 0
    return dels

File: iprep/strings/making_anagrams.scala
import java.io._
import java.math._
import java.security._
import java.text._
import java.util.concurrent._
import java.util.function._
import java.util.regex._
import java.util.stream._

import scala.collection.mutable.HashMap
import scala.collection.immutable.Set

object Solution {

    // Complete the makingAnagrams function below.
    def makingAnagrams(s1: String, s2: String): Int = {

        val s1Count : Map[Char,Int] = s1.groupBy(identity).mapValues(_.size)
        val s2Count : Map[Char,Int] = s2.groupBy(identity).mapValues(_.size)

        var edits = 0

        for ((k,v) <- s2Count) {
            edits += Math.max(0, v - s1Count.getOrElse(k, 0))
        }

        for ((k,v) <- s1Count) {
            edits += Math.max(0, v - s2Count.getOrElse(k, 0))
        }

        return edits

    }

    def main(args: Array[String]) {
        val stdin = scala.io.StdIn

        val printWriter = new PrintWriter(sys.env("OUTPUT_PATH"))

        val s1 = stdin.readLine

        val s2 = stdin.readLine

        val result = makingAnagrams(s1, s2)

        printWriter.println(result)

        printWriter.close()
    }
}
